doc/plugins.txt

```diff
@@ -0,0 +1,83 @@
+===============================
+Extending Sumatra with plug-ins
+===============================
+
+To support the wide diversity of scientific workflows, Sumatra aims to make it
+easy (or at least possible) to customize the way it works. The following
+components can be customized:
+
+- launching computations (local execution, distributed using MPI, batch jobs using SLURM, ...)
+- parameter file formats (JSON, YAML, ...)
+- executable programs/programming languages (Python, MATLAB, R, ...)
+- data stores (local disk, network storage, WebDAV, Dropbox, ...)
+- record stores (SQLite, PostgreSQL, web-based, ...)
+- output formats (plain text, HTML, LaTeX, JSON, ...)
+- version control systems (Git, Mercurial, Subversion, ...)
+
+
+Here is an example of wrapping the PHP command-line interface. (In itself, this gives little or
+no benefit, as Sumatra can automatically obtain version information from most command-line
+tools, but (a) it is a prerequisite if we wish to add dependency tracking, (b) it makes a
+nicely simple example of writing a plug-in!)
+
+.. todo:: explore Zenodo and Figshare as extension examples
+
+.. code-block:: python
+
+    from sumatra.core import registry
+    from sumatra.programs import Executable
+
+    class PHPcli(Executable):
+        name = "PHP"
+        executable_names = ('php',)
+        file_extensions = ('.php',)
+        default_executable_name = 'php'
+        requires_script = True
+
+    registry.register(PHPcli)
+
+To use this plug-in in a Sumatra project, it should be saved in a file (e.g. :file:`php.py`)
+that is on the Python module search path (for example, installed in :file:`site-packages` or
+added to the :envvar:`PYTHONPATH` environment variable), and then added to the project using:
+
+.. code-block:: bash
+
+    $ smt configure --add-plugin=php
```

Comment:
Would be nice if I could configure that globally for myself or the whole machine (that could be as in Git using `--global` or `--system`). If I'd have my own recordstore implementation for example, I most probably would like to have that defined globally.

From `users.py` I know that there have been ideas of a per-user configuration already before. What do you think of extending that for plugins?